src/timedelta_isoformat/__init__.py

```diff
@@ -162,6 +162,8 @@ def fromisoformat(cls, duration: str) -> "timedelta":
 
         :raises: `ValueError` with an explanatory message when parsing fails
         """
+        if not isinstance(duration, str):
```

Comment:
Could you change this into an `assert` statement?

As context: assertion statements do not result in any code operations when optimization is enabled (see [this Python documentation](https://docs.python.org/3/reference/simple_stmts.html#grammar-token-python-grammar-assert_stmt)).

Basically I have two environments in mind for this module:

- People who want to parse ISO-8601 durations in a straightforward way, and don't have a particular requirement for high performance code - for them, we can provide stronger levels of validation
- People who would like to use the code in environments that have demanding performance requirements, where each additional comparison operation in a loop or method must be weighed-up against the benefit it provides.  They're somewhat more responsible for ensuring that the input in the environment is acceptable for the module (but this should be stated in the README, to be honest).